### PR TITLE
(Feature) [GoodID] Register to RedTent

### PR DESCRIPTION
**Pull request by johnsmith-gooddollar**
# Description

- redtent endpoint
- basic validations
- utils/helpers
- unit tests
- verify user location + gender from credentials are valid for redtent
- verify videoFilename equal {account}.{ext} and exists on our redtent bucket
- if all valid call multiwallet registerRedtent(account)

About #468 

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

File: src/server/goodid/goodid-middleware.js [hunk 1]
```diff
@@ -252,7 +254,97 @@ export default function addGoodIDMiddleware(app: Router, utils, storage) {
...
+
+        if (countryCode !== 'NG') {
+          throw new Error('Failed to verify: allowed for accounts from Nigeria only')
+          // TODO: verify user gender - howto ? @sirpy
+        }
+
+        await utils.checkS3AccountVideo(videoFilename, account)
+
+        // TODO: uncomment once implemented at wallet side
+        // MultiWallet.registerRedtent(account)
```
Comment (sirpy) [inline src/server/goodid/goodid-middleware.js:341]: uncomment create a stub method in multiwallet

**Review by sirpy — CHANGES_REQUESTED** [review]

**Comment by johnsmith-gooddollar** [comment]
@sirpy we need permission to execute HeadObjectCommand to check existence. It retrieves metadata only, not the file data itself, so do not worry regarding privacy. Pls ask Redtent to make it for everyone or give us access key with this permission enabled. I cannot finish implementation without it

**Comment by sirpy** [comment]
@johnsmith-gooddollar try now

**Comment by johnsmith-gooddollar** [comment]
@sirpy still have error:
```
 console.log
    AccessDenied: Request has no authentication!
        at throwDefaultError (/Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@smithy/smithy-client/dist-cjs/index.js:838:20)
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@smithy/smithy-client/dist-cjs/index.js:847:5
        at de_CommandError (/Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/client-s3/dist-cjs/index.js:4756:14)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@smithy/middleware-serde/dist-cjs/index.js:35:20
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/middleware-signing/dist-cjs/index.js:225:18
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@smithy/middleware-retry/dist-cjs/index.js:320:38
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:97:20
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:120:14
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/middleware-logger/dist-cjs/index.js:33:22
        at GoodIDUtils.checkS3AccountVideo (/Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/src/server/goodid/utils.js:136:42)
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/src/server/goodid/goodid-middleware.js:336:9 {
      '$fault': 'client',
      '$metadata': {
        httpStatusCode: 403,
        requestId: 'YNAHTN1AC4JKME1Z',
        extendedRequestId: 'emncoJbawdyuiOkZR1QwMFuOapyeyP+1wclmB1VHGSPzGCgPeN3RuBXtasuidQZ5wtsOtqWFB84=',
        cfId: undefined,
        attempts: 1,
        totalRetryDelay: 0
      },
      Code: 'AccessDenied',
      RequestId: 'YNAHTN1AC4JKME1Z',
      HostId: 'emncoJbawdyuiOkZR1QwMFuOapyeyP+1wclmB1VHGSPzGCgPeN3RuBXtasuidQZ5wtsOtqWFB84='
    }
```
(via GetObjectAtributesCommand)

**or**
```
  console.log
    403: UnknownError
        at throwDefaultError (/Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@smithy/smithy-client/dist-cjs/index.js:838:20)
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@smithy/smithy-client/dist-cjs/index.js:847:5
        at de_CommandError (/Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/client-s3/dist-cjs/index.js:4756:14)
        at processTicksAndRejections (node:internal/process/task_queues:95:5)
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@smithy/middleware-serde/dist-cjs/index.js:35:20
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/middleware-signing/dist-cjs/index.js:225:18
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@smithy/middleware-retry/dist-cjs/index.js:320:38
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:97:20
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/middleware-sdk-s3/dist-cjs/index.js:120:14
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/node_modules/@aws-sdk/middleware-logger/dist-cjs/index.js:33:22
        at GoodIDUtils.checkS3AccountVideo (/Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/src/server/goodid/utils.js:136:42)
        at /Users/alexserdukov/Documents/Projects/GoodDollar/GoodServer/src/server/goodid/goodid-middleware.js:336:9 {
      '$fault': 'client',
      '$metadata': {
        httpStatusCode: 403,
        requestId: 'BA9ZDGHMWHEARNDQ',
        extendedRequestId: '8PfqNosLxrA3tTq8/szdxm58EHKdfm9w10x8BZVHOmumQ8nqyhC2DfXujI82eu+NxYkteW51Ui0=',
        cfId: undefined,
        attempts: 1,
        totalRetryDelay: 0
      }
    }
```
(via HeadObjectCommand)

**Comment by johnsmith-gooddollar** [comment]
@sirpy pls check out which permissions are needed: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/client/s3/command/HeadObjectCommand/

also should I use some authorisation ? at w3sdk example s3 client created without any auth keys....

**Comment by johnsmith-gooddollar** [comment]
@sirpy done

**Review by sirpy — APPROVED** [review]

**Pull request merged**